sway/tree/view.c

```diff
@@ -54,6 +54,7 @@ void view_destroy(struct sway_view *view) {
 	}
 	list_free(view->executed_criteria);
 
+	view->xdg_decoration->view = NULL;
```

Comment:
This needs a NULL check for `view->xdg_decoration` since not every view is going to have a `xdg_decoration`. Also, only `xdg_shell` views can have a `xdg_decoration` so it probably makes more sense to have this in `sway/desktop/xdg_shell.c`

Reply:
Yep, moved and added the check.